framework/src/ics/InitialConditionBase.C

```diff
@@ -52,6 +60,7 @@ InitialConditionBase::InitialConditionBase(const InputParameters & parameters)
     DependencyResolverInterface(),
     Restartable(this, "InitialConditionBases"),
     ElementIDInterface(this),
+    _my_state(getParam<int>("state")),
```

Comment:
```suggestion
    _my_state(getParam<MooseEnum>("state")),
```

The enum should automatically cast to int